### PR TITLE
Split main.py out of cli.py

### PR DESCRIPTION
**Pull request by pde**
The first part of a solution to #2442.

`renewer.py` and `plugins/selection.py` yet to come.

### CODE REVIEW

File: letsencrypt/cli.py [hunk 1]
```diff
@@ -1,20 +1,14 @@
-"""Let's Encrypt CLI."""
-from __future__ import print_function
-
-# TODO: Sanity check all input.  Be sure to avoid shell code etc...
+"""Let's Encrypt command CLI argument processing."""
```
Comment (bmw) [inline letsencrypt/cli.py]: s/command CLI/CLI

File: letsencrypt/cli.py [hunk 2]
```diff
@@ -1158,19 +629,16 @@ class HelpfulArgumentParser(object):
...
+    def __init__(self, args, plugins, detect_defaults=False):
 
-    # List of topics for which additional help can be provided
-    HELP_TOPICS = ["all", "security",
-                   "paths", "automation", "testing"] + list(six.iterkeys(VERBS))
+        from letsencrypt import main
+        self.VERBS = main.VERBS
+        # List of topics for which additional help can be provided
+        HELP_TOPICS = ["all", "security",
+                       "paths", "automation", "testing"] + list(six.iterkeys(self.VERBS))
```
Comment (bmw) [inline letsencrypt/cli.py]: nit: This can just be `list(self.VERBS)`. Works in all the Pythons.

File: letsencrypt/cli.py [hunk 3]
```diff
@@ -1158,19 +629,16 @@ class HelpfulArgumentParser(object):
...
 
-    def __init__(self, args, plugins, detect_defaults=False):
         plugin_names = list(six.iterkeys(plugins))
```
Comment (bmw) [inline letsencrypt/cli.py]: Same is true of this line `list(plugins)`, but I know using `iterkeys` wasn't introduced in this PR.

File: letsencrypt/main.py [hunk 4]
```diff
@@ -0,0 +1,716 @@
...
+
+from letsencrypt.cli import choose_configurator_plugins, _renewal_conf_files, should_renew
+from letsencrypt.display import util as display_util, ops as display_ops
+from letsencrypt.plugins import disco as plugins_disco
+
+import traceback
+import logging.handlers
+import time
+from acme import jose
+import OpenSSL
```
Comment (bmw) [inline letsencrypt/main.py]: nit: Plugin ordering. Order is:
```
import builtins 

import dependencies

import our own code
```
Each section is alphabetized. Doing this is consistent across the project.

File: letsencrypt/main.py [hunk 5]
```diff
@@ -0,0 +1,716 @@
+"""Let's Encrypt main entry point."""
+from __future__ import print_function
+import atexit
+import functools
+import os
+import sys
+import zope.component
```
Comment (bmw) [inline letsencrypt/main.py]: nit: Should be an empty line between `import sys` and this line.

File: letsencrypt/main.py (hunk 4 above)
Comment (pde) [inline letsencrypt/main.py]: Oops yes.  I only partially cleaned up after `python-rope` there...

**Comment by bmw** [comment]
@jsha: [serverInternal :: Error creating new authz](https://gist.github.com/92303b160a130939b695)

File: letsencrypt/main.py [hunk 6]
```diff
@@ -0,0 +1,717 @@
...
+from letsencrypt import configuration
+from letsencrypt import constants
+from letsencrypt import errors
+from letsencrypt import interfaces
+from letsencrypt import le_util
+from letsencrypt import log
+from letsencrypt import reporter
+from letsencrypt import storage
+
+from acme import jose
```
Comment (bmw) [inline letsencrypt/main.py]: Alphabetizing imports, this should be above `import letsencrypt` with a blank line on each side like in [master](https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/cli.py#L27).

File: letsencrypt/tests/display/util_test.py [hunk 7]
```diff
@@ -8,7 +8,6 @@
 
 from letsencrypt.display import util as display_util
 
-
```
Comment (bmw) [inline letsencrypt/tests/display/util_test.py]: super nit: This change does nothing but cause more `pep8` errors.

**Comment by bmw** [comment]
Other than my two tiny additional nits, my only other complaint is that I that `VERBS` should be in `cli` not `main` (as @pde pointed out). I created #2666 against this branch to make that work.

**Comment by bmw** [comment]
Great! Once the two nits above are fixed, I'll merge and move on to your next PR.

**Comment by bmw** [comment]
LGTM

**Pull request merged**